featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -53,50 +55,53 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
     """
     if not HAS_AZURE_MONITOR_EVENTS_EXTENSION:
         return
-    event: Dict[str, Optional[str]] = {}
-    if not evaluation_event.feature:
+
+    feature = evaluation_event.feature
+
+    if not feature:
         return
-    event[FEATURE_NAME] = evaluation_event.feature.name
-    event[ENABLED] = str(evaluation_event.enabled)
-    event["Version"] = EVALUATION_EVENT_VERSION
+
+    event: Dict[str, Optional[str]] = {
+        FEATURE_NAME: feature.name,
+        ENABLED: str(evaluation_event.enabled),
+        "Version": EVALUATION_EVENT_VERSION,
+    }
+
+    reason = evaluation_event.reason
+    variant = evaluation_event.variant
 
     # VariantAllocationPercentage
-    if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
-        if evaluation_event.variant:
-            event[VARIANT] = evaluation_event.variant.name
-        event[REASON] = evaluation_event.reason.value
-
-        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
-            allocation_percentage = 0
-
-            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
-                for allocation in evaluation_event.feature.allocation.percentile:
-                    if (
-                        evaluation_event.variant
-                        and allocation.variant == evaluation_event.variant.name
-                        and allocation.percentile_to
-                    ):
-                        allocation_percentage += allocation.percentile_to - allocation.percentile_from
+    if reason and reason != VariantAssignmentReason.NONE:
```

Comment:
I think this check can be removed. The variant should be None already so lines 75-77 will run correctly. The remaining logic has explicit checks that would also be skipped. And we should emit the Reason of None.

Reply:
Should there be a default reason if the reason object is `None`?

Reply:
I prefer VariantAssignmentReason be an enum with VariantAssignmentReason.NONE as the default value. I don't think it should be optional

Reply:
@rossgrambo. Sorry I mean `None` as in `null` not `VariantAssignmentReason.NONE`

Reply:
Yeah that was my understanding. In other languages- an enum can't be null and it defaults to the first value. I see that's not a thing in python- Perhaps at the top of this you can just do:

```python
if (!reason)
    reason = VariantAssignmentReason.NONE

othercode~
```

Reply:
We could solve this better by making so the default reason is `VariantAssignmentReason.NONE` and that would make it not `None`able.